app/controllers/profile_controller.rb

```diff
@@ -5,5 +5,7 @@ class ProfileController < ApplicationController
   def index
     cacher = Pii::Cacher.new(current_user, user_session)
     @decrypted_pii = cacher.fetch
+    @recovery_code = flash[:recovery_code]
+    flash.delete(:recovery_code)
```

Comment:
I thought so too, because hashes work that way. Alas, flash does not. It returns `self`.

https://github.com/rails/rails/blob/bf876aa0b61995f1be9b0146df7db74bd34d46af/actionpack/lib/action_dispatch/middleware/flash.rb#L177